### PR TITLE
refactor(TestCache): guarantee test caches are closed when tests finish

### PR DESCRIPTION
**Pull request by smoyer64**
Related to #978.

### CODE REVIEW

**Comment by smoyer64** [comment]
@zinderic - this is the follow-up to #978.  If the previous PR fixed the problems with GitHub builds (as it did on Mac OSX), this PR shouldn't behave any differently.

**Comment by zinderic** [comment]
@smoyer64 yeah, I just tested and I don't see failures on MacOS for this test with the refactored version in this PR.

**Review by MichaelMure — APPROVED** [review]

File: cache/repo_cache_test.go [hunk 1]
```diff
@@ -308,19 +313,26 @@ func checkBugPresence(t *testing.T, cache *RepoCache, bug *BugCache, presence bo
...
+	})
 
-	backend, err := NewRepoCacheNoEvents(repo)
-	require.NoError(t, err)
+	return cache
+}
 
-	i, err := backend.Identities().New("René Descartes", "rene@descartes.fr")
-	require.NoError(t, err)
+func indexCount(t *testing.T, repo repository.TestedRepo, name string) uint64 {
```
[the remark this answers is not in the document]
Comment (smoyer64) [reply cache/repo_cache_test.go:329]: I had considered that - it also means that `t` and `repo` can be used without passing them as arguments.  What I don't know is whether `t.Helper()` has an effect in a function declared in this manner.

File: cache/repo_cache_test.go [hunk 2]
```diff
@@ -18,106 +18,105 @@ import (
...
+		q, err = query.Parse("status:open marker") // full-text search
+		require.NoError(t, err)
+		res, err = cache.Bugs().Query(q)
+		require.NoError(t, err)
+		require.Len(t, res, 1)
+	})
 
-	// Close
-	require.NoError(t, cache.Close())
+	// Close (repository is closed by t.Cleanup() when the function above is exited)
```
[the remark this answers is not in the document]
Comment (smoyer64) [reply cache/repo_cache_test.go:101]: I had considered implicitly testing `Close()` by including it in the `t.Helper()` but that seemed wrong(er).

**Pull request merged**